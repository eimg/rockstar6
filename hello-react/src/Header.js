import React from 'react';
// import PropTypes from 'prop-types';

const Header= props => (
    <h1>Hello React {props.count}</h1>
);

// Header.propTypes = {
//     count: propTypes.number.isRequired
// };

export default Header;
