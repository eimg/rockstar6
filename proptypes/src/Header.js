import React from 'react';
import PropTypes from 'prop-types';

const Header = props => {
    return (
        <div>
            <h1 title="TDD">Prop Type {props.count}</h1>
        </div>
    )
};

Header.propTypes = {
    count: PropTypes.number.isRequired
}

export default Header;
