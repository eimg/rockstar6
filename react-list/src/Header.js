import React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Badge from "@material-ui/core/Badge";
import Button from "@material-ui/core/Button";
import ListIcon from "@material-ui/icons/List";

const Header = ({ count, clear }) => {
    return (
        <AppBar position="static">
            <Toolbar>
                <Badge badgeContent={count} color="secondary">
                    <ListIcon />
                </Badge>
                <Typography
                    style={{marginLeft: 30, flexGrow: 1}}
                    variant="h6">Todo List</Typography>
                <Button onClick={clear} color="inherit">CLEAR</Button>
            </Toolbar>
        </AppBar>
    )
}

export default Header;
