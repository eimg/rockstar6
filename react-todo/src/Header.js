import React from 'react';

const Header = ({ count }) => (
    <h1>React Todo ({count})</h1>
);

export default Header;
