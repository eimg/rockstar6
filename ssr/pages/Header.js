import React from 'react';

const style = {
    background: 'cyan',
    padding: 10
};

const Header = props => {
    return (
        <div style={style}>
            <h1>Next JS</h1>
            {props.children}
        </div>
    );
}

export default Header;
