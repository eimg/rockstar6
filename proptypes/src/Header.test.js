import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Header count={2} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
