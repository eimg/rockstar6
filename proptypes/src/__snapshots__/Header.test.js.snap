// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    title="TDD"
  >
    Prop Type 
    2
  </h1>
</div>
`;
